models/esm2/tests/test_convert_reverse.py

```diff
@@ -0,0 +1,139 @@
+        exported_keys = {k for k in exported_state_dict.keys() if "contact_head" not in k}
+        assert original_keys == exported_keys
+
+        for key in original_state_dict.keys():
+            if not key.endswith("_extra_state") and not key.endswith("inv_freq") and "contact_head" not in key:
+                torch.testing.assert_close(original_state_dict[key], exported_state_dict[key], atol=1e-5, rtol=1e-5)
+
+
+def test_qkv_unpacking():
+    """Test that QKV unpacking works correctly."""
+    from esm.convert import convert_esm_hf_to_te, convert_esm_te_to_hf
+
+    model_hf = AutoModelForMaskedLM.from_pretrained("facebook/esm2_t6_8M_UR50D")
+    model_te = convert_esm_hf_to_te(model_hf)
+    model_hf_converted = convert_esm_te_to_hf(model_te)
+
+    for i in range(model_hf.config.num_hidden_layers):
+        hf_query = model_hf.state_dict()[f"esm.encoder.layer.{i}.attention.self.query.weight"]
+        hf_key = model_hf.state_dict()[f"esm.encoder.layer.{i}.attention.self.key.weight"]
+        hf_value = model_hf.state_dict()[f"esm.encoder.layer.{i}.attention.self.value.weight"]
+
+        converted_query = model_hf_converted.state_dict()[f"esm.encoder.layer.{i}.attention.self.query.weight"]
+        converted_key = model_hf_converted.state_dict()[f"esm.encoder.layer.{i}.attention.self.key.weight"]
+        converted_value = model_hf_converted.state_dict()[f"esm.encoder.layer.{i}.attention.self.value.weight"]
+
+        torch.testing.assert_close(hf_query, converted_query, atol=1e-5, rtol=1e-5)
+        torch.testing.assert_close(hf_key, converted_key, atol=1e-5, rtol=1e-5)
+        torch.testing.assert_close(hf_value, converted_value, atol=1e-5, rtol=1e-5)
+
+
+def test_config_conversion():
+    """Test that config conversion works correctly."""
+    from esm.convert import convert_esm_hf_to_te, convert_esm_te_to_hf
+
+    model_hf = AutoModelForMaskedLM.from_pretrained("facebook/esm2_t6_8M_UR50D")
+    model_te = convert_esm_hf_to_te(model_hf)
+    model_hf_converted = convert_esm_te_to_hf(model_te)
+
+    assert model_hf_converted.config.model_type == "esm"
+    assert model_hf_converted.config.hidden_size == model_hf.config.hidden_size
+    assert model_hf_converted.config.num_hidden_layers == model_hf.config.num_hidden_layers
+    assert model_hf_converted.config.num_attention_heads == model_hf.config.num_attention_heads
+    assert model_hf_converted.config.intermediate_size == model_hf.config.intermediate_size
+    assert model_hf_converted.config.vocab_size == model_hf.config.vocab_size
+
+    # assert not hasattr(model_hf_converted.config, 'qkv_weight_interleaved')
+    # assert not hasattr(model_hf_converted.config, 'encoder_activation')
+    # assert not hasattr(model_hf_converted.config, 'attn_input_format')
+    # assert not hasattr(model_hf_converted.config, 'fuse_qkv_params')
+    # assert not hasattr(model_hf_converted.config, 'micro_batch_size')
+    # assert not hasattr(model_hf_converted.config, 'max_seq_length')
```

Comment:
any reason these are commented out?

Reply:
Thanks, forgot to uncomment that.  I rewrote this section